app/src/main/java/com/xa/xpensauditor/Day_wise_visualization.java

```diff
@@ -137,33 +141,54 @@ public void onDataChange(@NonNull DataSnapshot datasnapshot) {
                             System.out.println(" Finallyyy   "+amm);
                             int am1=Integer.valueOf(amm);
                             tot=am1+tot;
-      dumm=snapshot.child("Day").getValue().toString();
+                            dumm=snapshot.child("Day").getValue().toString();
+                            dayyy1=Integer.valueOf(dumm);
+                            Calendar c1 = Calendar.getInstance();
+                            c1.set(y2, m2-1, dayyy1, 0, 0);
+                            Date date = new GregorianCalendar(y2, m2-1, dayyy1).getTime();
+
+
       dumm1=Integer.valueOf(dumm);
-      System.out.println("dummmmm   "+dumm1+"    "+dayyy);
-      if(dayyy==dumm1){
+      System.out.println("dummmmm   "+dumm1+"    "+date);
+
+      if(sortedMap.containsKey(date)){
           System.out.println("inside");
           int temp=0;
-          temp=hm.get(dayyy);
+          temp=sortedMap.get(date);
       temp=temp+am1;
           System.out.println("dummmmm1   "+dayyy+"    "+temp);
-      hm.put(dayyy,temp);
+          sortedMap.put(date,temp);
+          System.out.println("Dates printing-4   ");
       }
       else{
-      dayy=snapshot.child("Day").getValue().toString();
-      dayyy=Integer.parseInt(dayy);
-     hm.put(dayyy,am1);}
+          sortedMap.put(date,am1);}
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>';' is not followed by whitespace.